src/granite_orm.cr

```diff
@@ -131,15 +131,19 @@ class Granite::ORM
 
     # keep a hash of the params that will be passed to the adapter.
     def params
-      params = [] of DB::Any
+      parsed_params = [] of DB::Any
       {% for name, type in FIELDS %}
-        params << {{name.id}}
+        {% if type.id == Time.id %}
+          parsed_params << {{name.id}}.try(&.to_s("%F %X"))
+        {% else %}
+          parsed_params << {{name.id}}
+        {% end %}
       {% end %}
       {% if SETTINGS[:timestamps] %}
-        params << created_at.not_nil!.to_s("%F %X")
-        params << updated_at.not_nil!.to_s("%F %X")
+        parsed_params << created_at.not_nil!.to_s("%F %X")
+        parsed_params << updated_at.not_nil!.to_s("%F %X")
```

Comment:
Why we can't disallow nil in `crated_at` and `updated_at` in type definition [here](https://github.com/amber-crystal/granite-orm/blob/c49ff76e5d25cbb8217d57b27469f5b54ad7152e/src/granite_orm.cr#L71-L72) and avoid `not_nil!` runtime check?